tests/queries/0_stateless/00398_url_functions.sql

```diff
@@ -8,6 +8,12 @@ SELECT protocol('//127.0.0.1:443/') AS Scheme;
 
 SELECT '====HOST====';
 SELECT domain('http://paul@www.example.com:80/') AS Host;
+SELECT domain('user:password@example.com:8080') AS Host;
+SELECT domain('http://user:password@example.com:8080') AS Host;
+SELECT domain('http://user:password@example.com:8080/path?query=value#fragment') AS Host;
+SELECT domain('newuser:@example.com') AS Host;
+SELECT domain('http://:pass@example.com') AS Host;
+SELECT domain(':newpass@example.com') AS Host;
```

Comment:
Consider adding examples with multiple `@`/`:`, i.e. `http://foo:bar@bar@.com`.
Now this implementation works like `urllib.urlparse` in python, however it is better to document this in the tests.